server/src/main/java/org/opensearch/cluster/service/ClusterManagerService.java

```diff
@@ -21,4 +23,8 @@ public class ClusterManagerService extends MasterService {
     public ClusterManagerService(Settings settings, ClusterSettings clusterSettings, ThreadPool threadPool) {
         super(settings, clusterSettings, threadPool);
     }
+
+    public ClusterManagerService(Settings settings, ClusterSettings clusterSettings, ThreadPool threadPool, LongSupplier nanoTimeSupplier) {
```

Comment:
can you add the caller which will call this new constructor

Reply:
why do we need supplier, any particular reason?

Reply:
> why do we need supplier, any particular reason?

1. Better abstraction - don't need to scavenge through entire class to make changes in future if required.
2. Ease of testing - else we will have to go override the private methods (ofc, by declaring them protected just for testing) in tests like for [ClusterApplierService](https://github.com/opensearch-project/OpenSearch/blob/aa2158555c7fea440d8de80b939a8c646aa5d415/server/src/main/java/org/opensearch/cluster/service/ClusterApplierService.java#L714) in [ClusterApplierServiceTests](https://github.com/opensearch-project/OpenSearch/blob/aa2158555c7fea440d8de80b939a8c646aa5d415/server/src/test/java/org/opensearch/cluster/service/ClusterApplierServiceTests.java#L626). So having a Supplier seemed a clean design choice.

Reply:
Resolving this conversation since we changed the implementation entirely to have a utility method inside `ThreadPool` class itself.